jupyterhub/apihandlers/base.py

```diff
@@ -65,10 +86,9 @@ def check_referer(self):
         return True
 
     def get_scope_filter(self, req_scope):
-        """Produce a filter for `*ListAPIHandlers* so that GET method knows which models to return
-        If the APIHandler has unrestricted access
-        """
-        kind_regex = re.compile(r':?(users|services|groups):?')
+        """Produce a filter for `*ListAPIHandlers* so that GET method knows which models to return.
+        Filter is a callable that takes a resource name and outputs true or false"""
+        kind_regex = re.compile(r':?(user|service|group)s:?')
```

Comment:
Minor: to avoid compiling a pattern on every call, we can define the pattern at a higher level - a class attribute or module constant